fix: build seed docker multiplatform

armandobelardo opened the pull request:
Ensure seed containers are built to run on diff platforms

armandobelardo committed:
build seed docker multiplatform

armandobelardo committed:
run on branch for testing

armandobelardo committed:
see if this is the right set of libs for multiplat

armandobelardo committed:
see if this is the right set of libs for multiplat

armandobelardo committed:
this works locally

armandobelardo committed:
try this workaround from github

armandobelardo committed:
try this other workaround from github

armandobelardo committed:
the previous workaround actually worked

armandobelardo committed:
working

armandobelardo committed:
fix, java: Address NPE for RequestOptions with new timeout feature (#3053)

Commit:
(fix): `fern check` no longer throws when an undiscriminated union is a list of primitives (#3055)

* (fix): handle undiscriminated unions that are primitive

* run fern check on test definitions when the definitions change

* fix names

* fix snapshots

* gitignore poery.locks for python seed snapshots

dsinghvi commented on .github/workflows/build-seed.yml:
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - ab/fixSeed
```
lets take this out before merging!

armandobelardo replied:
good catch! removed!

dsinghvi commented:
@armandobelardo it took 49m to build the seed container for python, should we just not make it multiplatform?

armandobelardo committed:
Update build-seed.yml

Commit:
(fix): OpenAPI/AsyncAPI importer handles invalid datetime examples (#3056)

(fix): OpenAPI/AsyncAPI importer handles invalid datetimes

armandobelardo commented:
> @armandobelardo it took 49m to build the seed container for python, should we just not make it multiplatform?

Yeah it's horrible, nothing I tried made it faster though

When it's not multiplatform we just can't run it locally, the container that's used for seed in Actions is different from the one we use on macs, so we need to build for both archs

We could change the base runners for the actions though so architectures match, I don't have much of a preference since this image shouldn't change

Danwakeem committed:
docs: Updated express generator ID (#3057)

dsinghvi commented:
> We could change the base runners for the actions though so architectures match, I don't have much of a preference since this image shouldn't change

Let's do this, 49m on the Seed Runner is quite expensive for Github actions

armandobelardo committed:
fix: ensure we apply audience-based filtering to examples as well (#3043)

* add in audiences to seed and allow passing config to custom fixtures

* remove these logs

* ensure customer test works

* cr

* update these unrelated tests

* update tests

armandobelardo committed:
feat, fern: allow headers to specify their envvar as well (#3061)

armandobelardo committed:
feat, python: support envvar scanning for headers (#3064)

amckinney committed:
(chore): Release fern-go-sdk 0.17.0 (#3066)

amckinney committed:
(feature): Go supports multiple files in upload (#3070)

dependabot[bot] committed:
build(deps): bump github.com/fern-api/generator-exec-go from 0.0.600 to 0.0.603 in /generators/go (#3063)

build(deps): bump github.com/fern-api/generator-exec-go

Bumps [github.com/fern-api/generator-exec-go](https://github.com/fern-api/generator-exec-go) from 0.0.600 to 0.0.603.
- [Release notes](https://github.com/fern-api/generator-exec-go/releases)
- [Commits](https://github.com/fern-api/generator-exec-go/compare/v0.0.600...v0.0.603)

---
updated-dependencies:
- dependency-name: github.com/fern-api/generator-exec-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
(feature): deep object serialization in TypeScript SDK (#3060)

* deep object serialization in typescript

* rerun seed

* fix seed.yml

* add changelog and version

dsinghvi committed:
(chore, typescript): release 0.12.0 of the typescript SDK

Commit:
(fix, typescript): serialize optional deep object query params correctly in the TypeScript SDK  (#3071)

* (fix): ts sdk properly serializes optional objects

* add changelog and version bump for ts sdk

* Update CHANGELOG.md

* fix test

armandobelardo committed:
fix, ruby: Ensure the name passed into the `X-Fern-SDK-Name` header is the name of the gem, not the client class (#3073)

Commit:
(fix, typescript): sdk code snippets dont render empty dicts for parameters with default values (#3074)

* (fix, typescript): sdk code snippets dont render empty dicts for parameters with default values

* update changelog

amckinney committed:
(chore): Refactor Pagination IR to support offset (#3072)

Commit:
(chore, internal): move `docs-config` to use local typescript sdk gen (#3047)

* docs-config -> local

* lint fix

* prettier

* monorepo lint

* prettier ignore

* fix

* fix compile

---------

Co-authored-by: dsinghvi <dsinghvi@umich.edu>

Commit:
(feature, beta): support reading `changelog` dir from api directory (#3075)

read changelog dir and send it to docs

dsinghvi committed:
(fix): set changelog pages to non empty

minaelee committed:
docs: multiple site layout and page updates (#3052)

* docs: multiple site layout, folder structure, and page updates

* docs: update links in README.md

* remove unnecessary file

* update page names in docs.yml, add introductory sentence to get-started.mdx, fix link in export-openapi.mdx

* add all updates since dec 2023 to new february changelog, add feb changelog to docs.yml, fix formatting in dec changelog

* add info about automatically generated h1 header for docs pages

* add fern docs to feb 2024 changelog

* add more slugs to reduce repetitive urls

---------

Co-authored-by: Danny Sheridan <danny@buildwithfern.com>

dannysheridan committed:
docs: overview diagram newer version (#3076)

swap out overview diagram

dannysheridan committed:
docs: use new overview diagram image  (#3077)

* swap out overview diagram

* add url for overview

minaelee committed:
docs: add info on new icon component (#3079)

minaelee committed:
docs: update availability documentation (#3078)

* docs: document x-fern-availability openapi extension

* docs: add details about how to use availability component

* docs: wording update

* docs: update desc of generally-available availability property behavior

Commit:
(feature): leverage OpenAPI extension `x-tags` for schemas (#3081)

(fix): leverage x-tags for schemas

dsinghvi committed:
(chore): support undiscriminated union example generation

armandobelardo committed:
fix: make express generator respect it's version while publishing (#3084)

armandobelardo committed:
fix, nit: update the name of the GH workflow step to match (#3085)

dsinghvi committed:
(fix): Java SDK generator uses UTF-8 Decoding for files

Commit:
(chore): disable  eslint `@typescript-eslint/no-unnecessary-condition`

armandobelardo committed:
fix: address recursive loop in example gen with a max depth and lookback (#3086)

fix recursive loop in example gen

dsinghvi committed:
try again

Commit:
(internal): stop running eslint (#3087)

stop running eslint

Commit:
(chore): upgrade mrlint and reenable eslint (#3088)

* reenable eslint

* fix

* god damn

armandobelardo committed:
fix: add missing ruby dependencies to ensure rubocop can install (#3090)

armandobelardo committed:
fix, ts: leverage the full package path for `reference.md` (#3083)

amckinney committed:
(feature): Add option to disable OpenAPI example generation (#3091)

Commit:
(ts, feature): introduce custom config for `tolerateRepublish` to re publish npm versions (#3093)

* (ts, feature): introduce  custom config

* fix compile

armandobelardo committed:
improvement, python: swap to literals instead of enums (#3082)

* add flag to swap to stremnums

* bump and changelog

* lint

* add a no config test for enums

armandobelardo committed:
Don't try to run if the matrix will be empty

For some reason this causes the action to fail

armandobelardo committed:
move to arm runners

armandobelardo committed:
try installing docker

armandobelardo committed:
try installing collima instead

armandobelardo committed:
try installing both?

armandobelardo committed:
try this workaround

armandobelardo committed:
revert back to ubuntu and just see if they'll run arm images

minaelee committed:
docs: add new sdks quickstarts and update docs.yml (#3095)

minaelee committed:
docs: update feb 2024 changelog (#3092)

Commit:
(fix): republish python seed container (#3098)

* Update Dockerfile.python

* fix

Commit:
(fix): support generating correct code snippets when extending base client in python (#3097)

* fix

* (fix): support generating accurate code snippets when writing to base_client.py

* specify changelog

* fix check

armandobelardo committed:
stop doing python-seed

armandobelardo committed:
Update build-seed.yml

dsinghvi committed:
(fix): lint now passes
